File: 200_code_python/14.py
# Viết bằng Python
# Tính gia thừa của một số: Nhập một số và tính gia thừa của nó.

# Nhập một số từ bàn phím
n = int(input("Nhap mot so: "))

# Tính gia thừa
gia_thua = 1
if n < 0:
    print("Khong co giai thua cua so am.")
elif n == 0:
    print("Gia thua cua o la 1.")
else:
    for i in range(1, n + 1):
        gia_thua *= i
    print("Gia thua cua " , n , " la", gia_thua)

File: 200_code_cpp/14.cpp
// Viết bằng C++
// Tính gia thừa của một số: Nhập một số và tính gia thừa của nó.
#include <iostream>
using namespace std;

int main() {
    // Khai báo biến 
    int n;
    unsigned long long giai_thua = 1;

    // Nhập một số từ bàn phím 
    cout << "Nhap mot so: ";
    cin >> n;

    // Tính giai thừa
    if (n < 0 ) {
        cout << "Khong có gia thua cho so am." << endl;
    } else if ( n == 0 || n ==1 ) {
        cout << "Gia thua cua " << n << " la 1." << endl;
    } else {
        for (int i = 1; i <= n; i++){
            giai_thua *= i;
        }
        cout << "Gia thua cua " << n << " la " << giai_thua << endl;
    }

    return 0;
}